specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,711 @@
+
+        /// Other command, any of the context menu items not falling in any of the enums above
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_OTHER,
+        
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR;
+    
+    /// Indicates the context selection type
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_TYPE
+    {
+        /// Indicates that the context menu was invoked on the page.
+        COREWEBVIEW2_CONTEXT_TYPE_PAGE,
+
+        /// Indicates that the context menu was invoked on a frame.
+        COREWEBVIEW2_CONTEXT_TYPE_FRAME,
+
+        /// Indicates that the context menu was invoked on a selection.
+        COREWEBVIEW2_CONTEXT_TYPE_SELECTION,
+
+        /// Indicates that the context menu was invoked on a link.
+        COREWEBVIEW2_CONTEXT_TYPE_LINK,
+
+        /// Indicates that the context menu was invoked on an editable component
+        COREWEBVIEW2_CONTEXT_TYPE_EDITABLE,
+        
+        /// Indicates that the context menu was invoked on audio.
+        COREWEBVIEW2_CONTEXT_TYPE_AUDIO,
+
+        /// Indicates that the context menu was invoked on an image.
+        COREWEBVIEW2_CONTEXT_TYPE_IMAGE,
+        
+        /// Indicates that the context menu was invoked on a video
+        COREWEBVIEW2_CONTEXT_TYPE_VIDEO,
+
+    } COREWEBVIEW2_CONTEXT_TYPE;
+
+    /// Indicates the menu item type
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND
+    {
+        /// A normal menu item kind, can be enabled or not.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL,
+
+        /// A checkbox. Can be checked or enabled
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_CHECKBOX,
+
+        /// A radio button type, can be checked or enabled
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO,
+
+        /// A separator type, used to visually differentiate sections of context menu items. Can not be disabled or checked but can be deleted from the list.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SEPARATOR,
+
+        /// A submenu type, will contain a list of its children
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SUBMENU,
+
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND;
+
+    /// Context Menu Items displayed by the Edge browser, holds the properties of a context menu item
+    [uuid(7aed49e3-a93f-497a-811c-749c6b6b6c65), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItem : IUnknown
+    {
+        /// Get the name displayed for the Context Menu Item, will already have been translated by the browser
+        [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+        /// Get the descriptor for the Context Menu Item
+        [propget] HRESULT Descriptor([out, retval] COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR* value);
+
+        /// Get the shortcut for the Context Menu Item, the functionality will already exist in the browser code
+        [propget] HRESULT Shortcut([out, retval] LPWSTR* value);
+
+        /// Get the Icon for the ContextMenuItem in the form of IStream
```

Comment:
We must also document what is the format of the image data in the IStream

Reply:
It seems like the Menu Model using ImageModel which uses Image and that will represent the icon in the form of a PNG. Will document that.